src/integrations/watchers/addon-update-watcher.php

```diff
@@ -0,0 +1,166 @@
+		return [ Admin_Conditional::class ];
+	}
+
+	/**
+	 * Replaces the auto-update toggle links for the Yoast add-ons
+	 * with a text explaining that toggling the Yoast SEO auto-update setting
+	 * automatically toggles the one for the setting for the add-ons as well.
+	 *
+	 * @param string $old_html The old HTML.
+	 * @param string $plugin   The plugin.
+	 *
+	 * @return string The new HTML, with the auto-update toggle link replaced.
+	 */
+	public function replace_auto_update_toggles_of_addons( $old_html, $plugin ) {
+		if ( ! \is_string( $old_html ) ) {
+			return $old_html;
+		}
+
+		if ( \in_array( $plugin, self::ADD_ONS, true ) ) {
+			$auto_updated_plugins = \get_option( 'auto_update_plugins' );
+
+			if ( $auto_updated_plugins === false || ! \is_array( $auto_updated_plugins ) ) {
+				return $old_html;
+			}
+
+			if ( $this->auto_updates_are_enabled_for_free( $auto_updated_plugins ) ) {
+				return \sprintf( '<em>%s</em>', \esc_html__( 'Auto-updates are enabled based on this setting for Yoast SEO.', 'wordpress-seo' ) );
+			}
+
+			return \sprintf( '<em>%s</em>', \esc_html__( 'Auto-updates are disabled based on this setting for Yoast SEO.', 'wordpress-seo' ) );
+		}
+
+		return $old_html;
+	}
+
+	/**
+	 * Enables premium auto updates when free are enabled and the other way around.
+	 *
+	 * @param array $old_value The old value of the `auto_update_plugins` option.
+	 * @param array $new_value The new value of the `auto_update_plugins` option.
+	 *
+	 * @return void
+	 */
+	public function toggle_auto_updates_for_add_ons_on_free_auto_update_toggle( $old_value, $new_value ) {
+		if ( ! \is_array( $old_value ) || ! \is_array( $new_value ) ) {
+			return;
+		}
+
+		if ( $this->auto_updates_for_free_have_just_been_enabled( $old_value, $new_value ) ) {
+			$this->enable_auto_updates_for_addons( $new_value );
+		}
+		elseif ( $this->auto_updates_for_free_have_just_been_disabled( $old_value, $new_value ) ) {
+			$this->disable_auto_updates_for_addons( $new_value );
+		}
+	}
+
+	/**
+	 * Checks whether auto updates for free have just been enabled.
+	 *
+	 * @param array $old_value The old value of the `auto_update_plugins` option.
+	 * @param array $new_value The new value of the `auto_update_plugins` option.
+	 *
+	 * @return bool Whether auto updates for free have just been enabled.
+	 */
+	protected function auto_updates_for_free_have_just_been_enabled( $old_value, $new_value ) {
+		return $this->auto_updates_are_enabled_for_free( $new_value ) && ! $this->auto_updates_are_enabled_for_free( $old_value );
+	}
+
+	/**
+	 * Checks whether auto updates for free have just been disabled.
+	 *
+	 * @param array $old_value The old value of the `auto_update_plugins` option.
+	 * @param array $new_value The new value of the `auto_update_plugins` option.
+	 *
+	 * @return bool Whether auto updates for free have just been disabled.
+	 */
+	protected function auto_updates_for_free_have_just_been_disabled( $old_value, $new_value ) {
```

Comment:
I would rename this to: `are_auto_updates_for_free_toggled`. This will makes the method reusable and you can use the outcome to take further action. By having this method you can use it for both cases where you swap the arguments you pass instead of having two method where you swap the passed arguments in the logic. 

In fact you are asking: 
- Were the auto updates enabled for value 1
- What is the current state of it in value 2.

Reply:
Andy and I looked at it and came up with a good solution together 🙂 .